readthedocs/api/v2/serializers.py

```diff
@@ -133,11 +136,49 @@ class Meta:
         exclude = []
 
 
+class BuildCommandUISerializer(BuildCommandSerializer):
+    """
+    Serializer used on GETs to trimm the commands' path.
+
+    Remove unreadable paths from the command outputs when returning it from the API.
+    We could make this change at build level, but we want to avoid undoable issues from now
+    and hack a small solution to fix the immediate problem.
+
+    This converts:
+        $ /usr/src/app/checkouts/readthedocs.org/user_builds/
+            <container_hash>/<project_slug>/envs/<version_slug>/bin/python
+        $ /home/docs/checkouts/readthedocs.org/user_builds/
+            <project_slug>/envs/<version_slug>/bin/python
+    into
+        $ python
+    """
+
+    command = serializers.SerializerMethodField()
+
+    def get_command(self, obj):
+        project_slug = obj.build.version.project.slug
+        version_slug = obj.build.version.slug
+        docroot = settings.DOCROOT.rstrip("/")  # remove trailing '/'
+
+        # Remove Docker hash from DOCROOT when running it locally
+        # DOCROOT contains the Docker container hash (e.g. b7703d1b5854).
+        # We have to remove it from the DOCROOT it self since it changes each time
+        # we spin up a new Docker instance locally.
+        container_hash = "/"
+        if settings.RTD_DOCKER_COMPOSE:
+            docroot = re.sub("/[0-9a-z]+/?$", "", settings.DOCROOT, count=1)
+            container_hash = "/[0-9a-z]+/"
+
+        regex = f"{docroot}{container_hash}{project_slug}/envs/{version_slug}(/bin/)?"
+        command = re.sub(regex, "", obj.command, count=1)
+        return command
+
+
 class BuildSerializer(serializers.ModelSerializer):
 
     """Build serializer for user display, doesn't display internal fields."""
 
-    commands = BuildCommandSerializer(many=True, read_only=True)
+    commands = BuildCommandUISerializer(many=True, read_only=True)
```

Comment:
We definitely need a comment or something explaining why this defaults to the UI serializer, instead of the regular one? It seems confusing to me we have:

BuildSerializer -> BuildAdminSerializer -> BuildAdminUISerializer

Why do we have 3 levels of nesting here, and what are each used for? It's not clear from the docstrings.